src/metabase/driver/mysql.clj

```diff
@@ -617,3 +620,59 @@
   [_driver]
   ;; https://dev.mysql.com/doc/refman/8.0/en/identifier-length.html
   64)
+
+(defn- sanitize-file-name
+  [name]
+  (str/replace name #"[^\w/\\.]" ""))
+
+(defn- format-load
+  [_clause [file-path table-name]]
+  [(format "LOAD DATA LOCAL INFILE '%s' INTO TABLE %s" (sanitize-file-name file-path) (sql/format-entity table-name))])
+
+(sql/register-clause! ::load format-load :insert-into)
+
+(defn- sanitize-value
+  ;; Per https://dev.mysql.com/doc/refman/8.0/en/load-data.html#load-data-field-line-handling
+  ;; Backslash is the MySQL escape character within strings in SQL statements. Thus, to specify a literal backslash,
+  ;; you must specify two backslashes for the value to be interpreted as a single backslash. The escape sequences
+  ;; '\t' and '\n' specify tab and newline characters, respectively.
+  [v]
+  (if (string? v)
+    (str/replace v #"\\|\n|\r|\t" {"\\" "\\\\"
+                                   "\n" "\\n"
+                                   "\r" "\\r"
+                                   "\t" "\\t"})
+    v))
+
+(defn- row->tsv
+  [row]
+  (->> row
+       (map sanitize-value)
+       (str/join "\t")))
+
+(defn- get-config
+  [db-id var-name]
+  (:value
+   (first
+    (jdbc/query (sql-jdbc.conn/db->pooled-connection-spec db-id)
+                ["show global variables like ?" var-name]))))
+
+(defmethod driver/insert-into :mysql
+  [driver db-id ^String table-name column-names values]
+  ;; `local_infile` must be turned on per
+  ;; https://dev.mysql.com/doc/refman/8.0/en/load-data.html#load-data-local
+  (if (not= (get-config db-id "local_infile") "ON")
+    ;; If it isn't turned on, fall back to the generic "INSERT INTO ..." way
+    (driver/insert-into :sql-jdbc db-id table-name column-names values)
+    (let [temp-file (File/createTempFile table-name ".tsv")
+          file-path (.getAbsolutePath temp-file)
+          tsv       (->> values
+                         (map row->tsv)
+                         (str/join "\n"))
+          sql       (sql/format {::load   [file-path (keyword table-name)]
+                                 :columns (map keyword column-names)}
+                                :quoted true
+                                :dialect (sql.qp/quote-style driver))]
```

Comment:
```suggestion
                                 :quoted true
                                 :dialect (sql.qp/quote-style driver))]
```

Reply:
I keep wanting to do that too, but they're not part of the map and so it's correct to align them with the `{` and not the `:`s in the map.